pkg/cmd/cli/cmd/newapp.go

```diff
@@ -90,6 +90,19 @@ application is created.`
 
   // Search for "ruby" in stored templates and print the output as an YAML
   $ %[1]s new-app --search --template=ruby --output=yaml`
+
+	newAppNoInput = `You must specify one or more images, image streams, templates or source code locations to create an application.
+
+Suggestions:
+Use '-L' or '--list' flag to list all local templates and image streams that can be used to create an application
```

Comment:
I still want to keep it very clear, remember that this is one of the central entry points for end-users. So I give you 13 words :)

```
You must specify one or more images, image streams, templates or source code locations to create an application.

To list all local templates and image streams, use

  $ oc new-app -L

To search templates, image streams, and Docker images that match the arguments provided, use

  $ oc new-app -S php
  $ oc new-app -S --template=ruby
  $ oc new-app -S --image=mysql
```